### PR TITLE
LP-2692 - Add validation error to a single field to improve UI

### PR DESCRIPTION
**Pull request by MunamMubashir-arbisoft**
[LP-2692](https://philanthropyu.atlassian.net/browse/LP-2692)

Previous PR: #336

### CODE REVIEW

File: openedx/adg/lms/applications/forms.py [hunk 1]
```diff
@@ -211,8 +211,6 @@ def clean(self):
         business_line_prerequisite = self.cleaned_data['business_line_prerequisite']
         is_common_business_line_prerequisite = self.cleaned_data['is_common_business_line_prerequisite']
 
-        if is_common_business_line_prerequisite and is_program_prerequisite:
+        if (business_line_prerequisite and (is_program_prerequisite or is_common_business_line_prerequisite)
+                or (is_common_business_line_prerequisite and is_program_prerequisite)):
             self.add_error('is_program_prerequisite', COURSE_GROUP_PREREQ_VALIDATION_ERROR)
-
```
Comment (javeriaejaz-arbisoft) [inline openedx/adg/lms/applications/forms.py:216]: If business_line_prerequisite and is_common_business_line_prerequisite are checked, then the error would be displayed on `is_program_prerequisite` field. 
Also, if you still want to go with this, you can also do
`if bool(business_line_prerequisite) + is_common_business_line_prerequisite + is_program_prerequisite > 1:`

**Review by javeriaejaz-arbisoft — CHANGES_REQUESTED** [review]

File: openedx/adg/lms/applications/forms.py [hunk 2]
```diff
@@ -211,8 +211,6 @@ def clean(self):
         business_line_prerequisite = self.cleaned_data['business_line_prerequisite']
         is_common_business_line_prerequisite = self.cleaned_data['is_common_business_line_prerequisite']
 
-        if is_common_business_line_prerequisite and is_program_prerequisite:
+        if (business_line_prerequisite and (is_program_prerequisite or is_common_business_line_prerequisite)
+                or (is_common_business_line_prerequisite and is_program_prerequisite)):
```
Comment (annas-usman) [inline openedx/adg/lms/applications/forms.py:215]:
```
(is_program_prerequisite or is_common_business_line_prerequisite) or (is_common_business_line_prerequisite and is_program_prerequisite)
```
will always return value equal to `(is_program_prerequisite or is_common_business_line_prerequisite)` so lets just remove `or (is_common_business_line_prerequisite and is_program_prerequisite)`

**Review by annas-usman — CHANGES_REQUESTED** [review]

File: openedx/adg/lms/applications/forms.py (hunk 1 above)
Comment (MunamMubashir-arbisoft) [reply openedx/adg/lms/applications/forms.py:216]: That is a great suggestion Javeria!

Comment (MunamMubashir-arbisoft) [reply openedx/adg/lms/applications/forms.py:216]: And yes, after discussing with QA, we only want the error displayed at the topmost field i.e `is_program_prerequisite`

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1915/workflows/workflows_id/jobs/1915/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/forms.py (100%)
-------------
Total:   1 line
Missing: 0 lines
Coverage: 100%
-------------
```

**Review by annas-usman — APPROVED** [review]

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1916/workflows/workflows_id/jobs/1916/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/forms.py (100%)
-------------
Total:   1 line
Missing: 0 lines
Coverage: 100%
-------------
```

**Pull request merged**